### PR TITLE
feat: edit note

### PR DESCRIPTION
**Pull request by manojncsoni007**
Description:
- User can edit their existing note

Screenshot:
image

### CODE REVIEW

File: src/service/addNewNote.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const addNewNote = async (noteObj, token, featureStateDispatch, setShowAddNote)
         featureStateDispatch({ type: "ADD_NOTES", payload: notes });
         setShowAddNote(false);
         showToast("success", "New note added")
```
Comment (atharvabhanage02) [inline src/service/addNewNote.js:15]: Good use of React Toastify

File: src/service/updateNote.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                { note: noteObj },
+                { headers: { authorization: token } })
+        featureStateDispatch({ type: "UPDATE_NOTES", payload: notes });
+        setShowAddNote(false);
+        showToast("success", "Note updated!!")
+        setEditNote(false);
+        setEditId("");
+        noteStateDispatch({ type: "RESET" })
+
+    } catch (error) {
```
Comment (atharvabhanage02) [inline src/service/updateNote.js:17]: You can use console.error here .

File: src/context/index.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 export {AuthProvider} from './auth-context'
 export {useAuth} from './auth-context'
 export {FeatureProvider} from './feature-context'
-export {useFeature} from './feature-context'
\ No newline at end of file
+export {useFeature} from './feature-context'
```
Comment (atharvabhanage02) [inline src/context/index.js:4]: Liked the way you are exporting functions and keeping the code maintainable

**Review by atharvabhanage02 — APPROVED** [review]
- UI looks great.
- Edit functionality is working fine
- Code is clean and well structured
Everything looks fine .

Suggestion : 
- If there are no Archived notes in Archive Page then  you can have a heading mentioning that

**Review by HeyNitin — APPROVED** [review]

File: src/service/updateNote.js (hunk 2 above)
Comment (manojncsoni007) [reply src/service/updateNote.js:17]: ok

**Pull request merged**